packages/shopping/src/services/basic.authorizor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  AuthorizationContext,
+  AuthorizationMetadata,
+  AuthorizationDecision,
+} from '@loopback/authorization';
+import _ from 'lodash';
+import {UserProfile, securityId} from '@loopback/security';
+
+interface MyAuthorizationMetadata extends AuthorizationMetadata {
+  currentUser?: UserProfile;
+  decision?: AuthorizationDecision;
+}
+
+// Instance level authorizer
+// Can be also registered as an authorizer, depends on users' need.
+export async function basicAuthorization(
```

Comment:
Basic authorization is an example of an authorization implementation. It implements the following:

1. `@authorize()` any controller method to enforce authorization, else it will be accessible to all.
2. Users with `admin` or `support` roles can access any controller method, IF the role was included in `@authorize({allowedRoles: [...]`.
3. To allow access only to a customer include only `customer` in `allowedRoles`. Now even `admin` and `support` cannot access these methods.